### DISCUSSION TITLE
Move type exports to `radashi/types` module.

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#144] [Ideas]
This RFC explores the idea of exporting **only functions** from the `radashi` main entry point. If you need types, import them from `radashi/types`. 

### Motivation
Most of the time, you don't need the types, so it's better to exclude them from the autocompletion dropdown in IDEs.

Minor detail: This change also encourages using `import type {…}` instead of `import { type Foo }`.

**Comment by MarlonPassos-git** [comment]
In my view, this proposal makes a lot of sense, especially considering that some of the types currently exported are not directly used by the functions but rather for constructing other types. Examples of this include `ExtractNotAny`, `SwitchAny`, `SwitchNever`, `BoxedPrimitive`, among others. Since the focus of `radashi` is to be a functional library and not a utility types library, it makes even more sense not to pollute the main import with unnecessary types.
